hw4/routes/index.js

```diff
@@ -0,0 +1,99 @@
+var mongoose = require('mongoose');
+var Ingredients = require('./../models/ingredientmodel');
+var routes = {};
+
+routes.home = function(req, res) {
+	res.render('home');
+};
+
+routes.edit = function(req, res) {
+	var updated = req.body;
+	Ingredients.update({'_id':updated.id}, updated, function(err, num, data) {
+		res.end(JSON.stringify(updated));
+	});
+};
+
+routes.ingredients = function(req, res) {
+	Ingredients.find({}, function(err, data) {
+		if (err) {
+			console.error(err);
+		}
+
+		var instock = [];
+		var outofstock = [];
+
+		data.forEach(function(data) {
+			if (data.stock) {
+				instock.push(data);
+			} else {
+				outofstock.push(data);
+			}
+		});
+
+		res.render('ingredients', 
+			{'instock': formatPrice(instock), 
+			'outofstock': formatPrice(outofstock)});
+		});
+};
+
+routes.outofstock = function(req, res) {
+// mark an ingredient out of stock
+	var ingredientId = req.body.id;
+	Ingredients.update({'_id':ingredientId}, {'stock':false}, function(err, num, data) {
+		res.end(ingredientId);
```

Comment:
Same thing here and for all the routes below. Whoever sends a request expects a response, and you're not giving it to them. Even if there's an error or something went wrong, you should still be sending them something back, even if it's just a short message letting them know there was an error. Don't leave them hanging!